### DISCUSSION TITLE
Naming format of video stream name causes API request to fail

### DISCUSSION THREAD

**Discussion by shanglike** [bluenviron/mediamtx#2972] [Questions]
### Question

When performing video streaming, the request is made using the 'http://localhost:9997/v3/config/paths/add/{name} 'method, with the request content being {"source": "rtsp://* * *: * * * @ 123 ^ @ 192.168.107.186"}. The final result obtained is {"error": "rtsp:/* * *: * * * @ 123 ^ @ 192.168.107.186' is not a valid URL"}.

I personally believe that the presence of '@ 123 ^' in the video stream address resulted in the IP address being judged as' @ 123 ^ @ 192.168.107.186 ', which ultimately led to this situation. Is there any other way to solve this situation without modifying the video stream address.

**Comment by shanglike** [comment]
The problem is caused by the presence of the ^ symbol in the video stream address.

**Comment by aler9** [comment]
moved to #2974

[a comment by an automated account is not shown]
